app/src/main/java/ro/code4/deurgenta/helper/MapViewUtils.kt

```diff
@@ -0,0 +1,182 @@
+import android.util.Log
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.here.sdk.core.*
+import com.here.sdk.mapview.*
+import com.here.sdk.search.*
+import ro.code4.deurgenta.R
+import ro.code4.deurgenta.ui.address.ConfigureAddressFragment
+
+class MapViewUtils(
+    private val mapView: MapView,
+    private val fusedLocationClient: FusedLocationProviderClient,
+    private val callback: MapViewCallback
+) {
+
+    private val mapMarkerList = mutableListOf<MapMarker>()
+    private val searchEngine: SearchEngine = SearchEngine()
+    private lateinit var lastGeoCoordinates: GeoCoordinates
+
+    private fun loadMapScene(location: Location, showLocation: Boolean) {
+        mapView.mapScene
+            .loadScene(MapScheme.NORMAL_DAY) { errorCode ->
+                if (errorCode == null) {
+
+                    val distanceInMeters = DEFAULT_DISTANCE_IN_METERS.toDouble()
+                    val geoCoordinates = GeoCoordinates(location.latitude, location.longitude)
+
+                    mapView.camera.lookAt(geoCoordinates, distanceInMeters)
+                    lastGeoCoordinates = geoCoordinates
+
+                    if (showLocation) {
+                        createAndAddMarker(geoCoordinates)
+                    }
+                } else {
+                    Log.e(TAG, "error loading results:$errorCode")
+                    if (errorCode != MapError.OPERATION_IN_PROGRESS) {
+                        callback.onError("cannot load map. Error:$errorCode")
+                    }
+                }
+            }
+    }
+
+    private fun createAndAddMarker(geoCoordinates: GeoCoordinates) {
+        val mapMarker = createPoiMapMarker(geoCoordinates)
+        mapView.mapScene.addMapMarker(mapMarker)
+        mapMarkerList.add(mapMarker)
+        callback.onAddMarker()
+    }
+
+    private fun getMapViewGeoBox(): GeoBox {
+        val mapViewWidthInPixels = mapView.width
+        val mapViewHeightInPixels = mapView.height
+        val bottomLeftPoint2D = Point2D(0.0, mapViewHeightInPixels.toDouble())
+        val topRightPoint2D = Point2D(mapViewWidthInPixels.toDouble(), 0.0)
+        val southWestCorner = mapView.viewToGeoCoordinates(bottomLeftPoint2D)
+        val northEastCorner = mapView.viewToGeoCoordinates(topRightPoint2D)
+        if (southWestCorner == null || northEastCorner == null) {
+            throw RuntimeException("GeoBox creation failed, corners are null.")
+        }
+
+        // Note: This algorithm assumes an unrotated map view.
+        return GeoBox(southWestCorner, northEastCorner)
+    }
+
+    fun searchOnMap(queryString: String) {
+        clearMap()
+        if (queryString.length < 2) {
+            return
+        }
+
+        val viewportGeoBox: GeoBox = getMapViewGeoBox()
+        val query = TextQuery(queryString, viewportGeoBox)
+
+        val maxItems = DEFAULT_MAX_ITEMS
+        val searchOptions = SearchOptions(LanguageCode.EN_GB, maxItems)
```

Comment:
you could use DEFAULT_MAX_ITEMS directly here, no need for extra var

also, shouldn't the search language be set to ro?

Reply:
Probably, but not sure it actually matters. I am searching in Austria, no difference seen. I will do some tests with country related code.